src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -857,16 +890,17 @@ public TaskWrapper workerIdle(Worker worker) {
                 long taskDelay = nextTask.getScheduleDelay();
                 if (taskDelay > 0) {
                   if (taskDelay == Long.MAX_VALUE) {
-                    // TODO - this will cause a spin if there is only one recurring task
-                    // don't sleep forever, this task is just starting to execute, so spin
-                    Thread.yield();
+                    // the hack at construction/start is to avoid this from causing us to spin here 
+                    // if only one recurring task is scheduled (otherwise we would keep pulling 
+                    // that task while it's running)
+                    continue;
                   }
                   if (queued) {
-                    if (! workerTimedPark) {
+                    if (nextTask.getPureRunTime() < workerTimedParkRunTime) {
```

Comment:
Was there an issue with just not doing this check?  Seems like we could always end up with a few threads in here anyway.